frontend_tests/node_tests/stream_data.js

```diff
@@ -509,14 +524,138 @@ run_test('stream_settings', () => {
     assert.equal(sub.stream_post_policy,
                  stream_data.stream_post_policy_values.everyone.code);
 
-    // For guest user only retrieve subscribed streams
+    // For guest user retrieve subscribed, unsubscribed and never subscribed
+    // streams. The unsubscribed streams are shown but unless they are web
```

Comment:
We could reword this to be:

"For guest users, retrieve:
- subscribed streams
- previously subscribed streams
- all web_public streams"